nodes/sros/sros.go

```diff
@@ -0,0 +1,1075 @@
+// Copyright 2025 Nokia
+// Licensed under the BSD 3-Clause License.
+// SPDX-License-Identifier: BSD-3-Clause
+
+package sros
+
+import (
+	"bytes"
+	"context"
+	_ "embed"
+	"fmt"
+	"io"
+	"maps"
+	"net"
+	"os"
+	"path"
+	"path/filepath"
+	"regexp"
+	"strconv"
+	"strings"
+	"text/template"
+	"time"
+
+	"github.com/charmbracelet/log"
+
+	"golang.org/x/crypto/ssh"
+
+	"github.com/srl-labs/containerlab/cert"
+	"github.com/srl-labs/containerlab/clab/exec"
+	"github.com/srl-labs/containerlab/labels"
+	"github.com/srl-labs/containerlab/netconf"
+	"github.com/srl-labs/containerlab/nodes"
+	"github.com/srl-labs/containerlab/nodes/state"
+	"github.com/srl-labs/containerlab/runtime"
+	"github.com/srl-labs/containerlab/types"
+	"github.com/srl-labs/containerlab/utils"
+)
+
+const (
+	SrosDefaultType = "SR-1" // default sros node type
+
+	readyTimeout = time.Minute * 1 // max wait time for node to boot
+
+	generateable     = true
+	generateIfFormat = "%d/%d/%d"
+
+	retryTimer = 1 * time.Second
+	// additional config that clab adds on top of the factory config.
+	scrapliPlatformName       = "nokia_sros"
+	configStartup             = "config/startup"
+	configCf3                 = "config/cf3"
+	configCf2                 = "config/cf2"
+	configCf1                 = "config/cf1"
+	startupCfgFName           = "config.cfg"
+	envNokiaSrosSlot          = "NOKIA_SROS_SLOT"
+	envNokiaSrosChassis       = "NOKIA_SROS_CHASSIS"
+	envNokiaSrosSystemBaseMac = "NOKIA_SROS_SYSTEM_BASE_MAC"
+	envNokiaSrosCard          = "NOKIA_SROS_CARD"
+)
+
+var (
+
+	//go:embed sros_default_config.go.tpl
+	srosConfigCmdsTpl string
+	kindNames         = []string{"nokia_srsim"}
+	srosSysctl        = map[string]string{
+		"net.ipv4.ip_forward":                "0",
+		"net.ipv6.conf.all.disable_ipv6":     "0",
+		"net.ipv6.conf.default.disable_ipv6": "0",
+		"net.ipv6.conf.all.accept_dad":       "0",
+		"net.ipv6.conf.default.accept_dad":   "0",
+		"net.ipv6.conf.all.autoconf":         "0",
+		"net.ipv6.conf.default.autoconf":     "0",
+		"net.ipv4.conf.all.rp_filter":        "0",
+		"net.ipv6.conf.all.accept_ra":        "0",
+		"net.ipv6.conf.default.accept_ra":    "0",
+		"net.ipv4.conf.default.rp_filter":    "0",
+	}
+	defaultCredentials = nodes.NewCredentials("admin", "NokiaSros1!")
+
+	srosEnv = map[string]string{
+		"SRSIM":                   "1",
+		envNokiaSrosChassis:       SrosDefaultType,     // fillers to be override
+		envNokiaSrosSystemBaseMac: "fa:ac:ff:ff:10:00", // filler to be override
+		envNokiaSrosSlot:          "A",                 // filler to be override
+	}
+
+	readyCmdCpm  = `/usr/bin/pidof cpm`
```

Comment:
Change from using `pidof` to `pgrep` to avoid issues with Rosetta emulation (`pgrep ^cpm$` and so on...)